tools/shared.py

```diff
@@ -1929,15 +1929,15 @@ def link_lld(args, target, opts=[], lto_level=0):
 
     if Settings.LINKABLE:
       cmd.append('--export-all')
+    else:
+      # in standalone mode, crt1 will call the constructors from inside the wasm
+      if not Settings.STANDALONE_WASM:
+        cmd += ['--export', '__wasm_call_ctors']
 
-    # in standalone mode, crt1 will call the constructors from inside the wasm
-    if not Settings.STANDALONE_WASM:
-      cmd += ['--export', '__wasm_call_ctors']
-
-    cmd += ['--export', '__data_end']
+      cmd += ['--export', '__data_end']
```

Comment:
I understand it's ok to put standalone wasm in the else of LINKABLE, as we don't support linking with that mode. And exported_functions is ok to put in that else because it isn't needed when linkable. However, I'm not sure what this `__data_end` thing is?

Reply:
Basically because LINKABLE added `--export-all` there is no point in having any `--export=` flag individually.   Its redundant to export all then then also export something individually in wasm-ld.

It is more link:

```
if exporting_everthing:
   add_export_all
else:
   add_export_specific
```